auth/composite_authenticator.go

```diff
@@ -1,26 +1,18 @@
 package auth
 
-type CompositeAuthenticator struct {
-	authenticators []Authenticator
-}
-
-func NewCompositeAuthenticator(authenticators ...Authenticator) *CompositeAuthenticator {
-	return &CompositeAuthenticator{
-		authenticators: authenticators,
-	}
-}
-
-func (c *CompositeAuthenticator) Authenticate(user, pass string) (bool, error) {
-	for _, authenticator := range c.authenticators {
-		isValid, err := authenticator.Authenticate(user, pass)
-		if err != nil {
-			return false, err
-		}
+func NewCompositeAuthenticator(authenticators ...Authenticator) AuthenticatorFunc {
+	return AuthenticatorFunc(func(user, pass string) (bool, error) {
+		for _, authenticator := range authenticators {
+			isValid, err := authenticator.Authenticate(user, pass)
+			if err != nil {
```

Comment:
We want to check which of the following scenarios an authenticator returned: 
* `true, nil` = the user sent good credentials for this particular authenticator
* `false, nil` = the user sent bad credentials for this particular authenticator
* `<anything>, error` = something unexpected went wrong with the authenticator

If we ever encounter the last scenario, we want to fail fast.